### PR TITLE
net-ssh 4.1

### PR DESCRIPTION
**Pull request by crami**
My try to get to the newest net-ssh version, as it supports more cyphers Im able to use ssh on more nodes. I tested my changes on  a few devices and that looks good. What is untested is the ssh proxy feature. Is someone able to verify this?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -508,7 +508,7 @@ source:
     map:
       name: ip
       model: model
-      username: username
+      user: username
```
Comment (ytti) [inline README.md]: Why this? node.rb uses ```username```

File: lib/oxidized/input/ssh.rb [hunk 2]
```diff
@@ -24,20 +24,22 @@ def connect node
...
       }
+
+      if proxy_host = vars(:ssh_proxy)
+        proxy_command =  "ssh "
+        proxy_command += "-o StrictHostKeyChecking=no " unless secure
+        proxy_command += "#{proxy_host} -W %h:%p"
+        proxy =  Net::SSH::Proxy::Command.new(proxy_command)
+        ssh_opts[:proxy] = proxy
+      end
+
```
Comment (ytti) [inline lib/oxidized/input/ssh.rb]: Why did you change this block from the current? Both are functional, seems unnecessary.

File: lib/oxidized/model/ironware.rb [hunk 3]
```diff
@@ -67,7 +67,7 @@ class IronWare < Oxidized::Model
     # match expected prompts on both older and newer
     # versions of IronWare
     username /^(Please Enter Login Name|Username):/
-    password /^(Please Enter Password |Password):/
+    password /^(Please Enter Password ?|Password):/
```
Comment (ytti) [inline lib/oxidized/model/ironware.rb]: This really should be in another PR.

File: README.md (hunk 1 above)
Comment (crami) [inline README.md]: Your are righ, changed it at the wrong location ... sql uses user not username...

File: lib/oxidized/model/ironware.rb (hunk 3 above)
Comment (crami) [inline lib/oxidized/model/ironware.rb]: Right ... can you tell me how to move that to the other pull request ?

File: lib/oxidized/input/ssh.rb (hunk 2 above)
Comment (crami) [inline lib/oxidized/input/ssh.rb]: No, they are not the same. In old version :proxy gets no value, which the ssh module complains about, in my version :proxy only gets set if there is need for a proxy.

**Comment by ytti** [comment]
I have another pull request for net-ssh 4.1 So maybe just scratch this PR, and create another with just the model change. 
Very helpful that you've arrived to same conclusion about the ssh_proxy!

Thank you.
